include/dca/phys/dca_step/cluster_solver/ctint/walker/ctint_walker_choice.hpp

```diff
@@ -0,0 +1,60 @@
+// Copyright (C) 2018 ETH Zurich
+// Copyright (C) 2018 UT-Battelle, LLC
+// All rights reserved.
+//
+// See LICENSE.txt for terms of usage.
+//  See CITATION.md for citation guidelines, if DCA++ is used for scientific publications.
+//
+// Author: Giovanni Balduzzi (gbalduzz@itp.phys.ethz.ch)
+//
+// Selects the appropriate ctint walker implementation.
+
+#ifndef DCA_PHYS_DCA_STEP_CLUSTER_SOLVER_CTINT_WALKER_CTINT_WALKER_CHOICE_HPP
+#define DCA_PHYS_DCA_STEP_CLUSTER_SOLVER_CTINT_WALKER_CTINT_WALKER_CHOICE_HPP
+
+#include "dca/linalg/linalg.hpp"
+#include "dca/phys/dca_step/cluster_solver/ctint/walker/ctint_walker_cpu.hpp"
+#include "dca/phys/dca_step/cluster_solver/ctint/walker/ctint_walker_cpu_submatrix.hpp"
+#ifdef DCA_HAVE_CUDA
+#include "dca/phys/dca_step/cluster_solver/ctint/walker/ctint_walker_gpu_submatrix.hpp"
+#endif  // DCA_HAVE_CUDA
+
+namespace dca {
+namespace phys {
+namespace solver {
+namespace ctint {
+// dca::phys::solver::ctint::
+
+namespace {
+template <linalg::DeviceType device, class Parameters, bool use_submatrix, typename Real>
+struct CtintWalkerChoicheSelector;
```

Comment:
I think this name is a typo or a joke I'm too old for.

Reply:
typo: fixed in cc35a74f7339b70d7bfc72c2a11d34a6ade06faa